releases/release-5.0.4.md

```diff
@@ -0,0 +1,165 @@
+---
+title: TiDB 5.0.4 Release Notes
+---
+
+# TiDB 5.0.4 Release Notes
+
+Release Date: September 27, 2021
+
+TiDB version: 5.0.4
+
+## Compatibility Changes
+
++ TiDB
+
+    - Revert #19341 to avoid #24326 [#26258](https://github.com/pingcap/tidb/pull/26258)
+
+## Improvements
+
++ TiDB
+
+    - Trigger auto-analyze based on histogram row count [#26707](https://github.com/pingcap/tidb/pull/26707)
+
++ TiKV
+
+    - separate read write ready to reduce read latency [#10620](https://github.com/tikv/tikv/pull/10620)
+
+## Bug Fixes
+
++ TiDB
+
+    - tidb panic while query hash partition table with is null condition [#26963](https://github.com/pingcap/tidb/pull/26963)
```

Comment:
- TiDB
  - tidb panic while query hash partition table with is null condition [#26963](https://github.com/pingcap/tidb/pull/26963)
  - Keep the overflow check logic same with mysql [#26724](https://github.com/pingcap/tidb/pull/26724)
  - Fix wrong charset and collation for case when function [#26672](https://github.com/pingcap/tidb/pull/26672)
  - Fix the issue that committing pessimistic transactions may report write-conflict errors. [#25974](https://github.com/pingcap/tidb/pull/25974)
  - Fix the issue that point get does not use lite version resolve lock [#26560](https://github.com/pingcap/tidb/pull/26560)
  - Fix the bug that index keys in a pessimistic transaction may be repeatedly committed. [#26495](https://github.com/pingcap/tidb/pull/26495)
  - Fix the issue that TiDB may panic when resolving async-commit locks. [#25863](https://github.com/pingcap/tidb/pull/25863)
  - check filter condition in func convertToPartialTableScan [#25806](https://github.com/pingcap/tidb/pull/25806)
  - Important security issue for handling ALTER USER statements [#25348](https://github.com/pingcap/tidb/pull/25348)
  - Fix a bug that a new cluster's "tidb_gc_scan_lock_mode" global variable shows "PHYSICAL" instead of the actual default mode "LEGACY". [#25118](https://github.com/pingcap/tidb/pull/25118)
  - Fix the bug that TIKV_REGION_PEERS table did not have the correct DOWN status. [#24919](https://github.com/pingcap/tidb/pull/24919)
  - Handle a potential statistic object's memory leak when HTTP api is used [#24649](https://github.com/pingcap/tidb/pull/24649)
  - SQL Views now consider the default roles associated with the SQL DEFINER correctrly. [#24532](https://github.com/pingcap/tidb/pull/24532)
  - Set the return value to 0 if --help is passed to cli, for golang below version 1.15 [#24074](https://github.com/pingcap/tidb/pull/24074)
  - fix dml_batch_size doesn't load the global variable [#24731](https://github.com/pingcap/tidb/pull/24731)
  - filter conflict read_from_storage hints [#24374](https://github.com/pingcap/tidb/pull/24374)
  - fix wrong type infer for agg function when type is null. [#24354](https://github.com/pingcap/tidb/pull/24354)
  - fix some potential panic in statistics [#24061](https://github.com/pingcap/tidb/pull/24061)
  - fix approx_percent panic on bit column [#23703](https://github.com/pingcap/tidb/pull/23703)

Reply:
Note: overwrite the original release notes in this PR.